pkg/component/shared/istio.go

```
@@ -263,6 +264,37 @@ func IsZonalIstioExtension(labels map[string]string) (bool, string) {
 	return false, ""
 }
 
+// AreZonalGatewaysInUse checks whether any shoots are using zonal Istio ingress gateways.
+// It returns true if any shoot control plane gateway targets a zonal Istio ingress gateway in the specified zones.
+func AreZonalGatewaysInUse(ctx context.Context, c client.Client, zones []string) bool {
+	gatewayList := &istionetworkingv1beta1.GatewayList{}
+	if err := c.List(ctx, gatewayList); err != nil {
+		// If Gateways can't be listed, return true
+		return true
```

Please return an error in this case. Otherwise, the Gateways might be deployed and deleted every now and then if the connection to the api-server is not stable. This is unlikely but it would be very hard to debug.

---

That is a good point, done in https://github.com/gardener/gardener/pull/12956/commits/b521c0d6a5a3985fea7ba0b838811f7ef82df372.